app/controllers/search_controller.rb

```diff
@@ -1,5 +1,5 @@
 class SearchController < ApplicationController
   def new
-    @results = Resource.search(params[:query])
+    @results = params[:query].present? ? Resource.search(params[:query]) : []
```

Comment:
The alternative is to more declaratively just call `Resource.search(params[:query])` and then over in the model, something like

```ruby
def self.search(query)
  if query
    super(query)
  else
    NullQuery.new.search # => []
  end
end